src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java

```diff
@@ -0,0 +1,137 @@
+package org.opendatakit.briefcase.model.form;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static java.util.stream.Collectors.toMap;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.delete;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.walk;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.write;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import org.opendatakit.briefcase.export.XmlElement;
+import org.opendatakit.briefcase.model.FormKey;
+import org.opendatakit.briefcase.reused.BriefcaseException;
+import org.opendatakit.briefcase.reused.LegacyPrefs;
+
+public class FileSystemFormMetadataAdapter implements FormMetadataPort {
```

Comment:
It would help the reviewer if these types had doc comments, even very brief ones.